xesmf/frontend.py

```diff
@@ -1221,6 +1209,23 @@ def __init__(
             unmapped_to_nan=False,
         )
 
+    @staticmethod
+    def _check_polys_length(polys):
+        # Check length of polys segments, issue warning if too long
+        check_polys, check_holes, _, _ = split_polygons_and_holes(polys)
+        check_polys.extend(check_holes)
+        poly_segments = []
+        for check_poly in check_polys:
+            b = check_poly.boundary.coords
+            poly_segments.extend(
+                [LineString(b[k : k + 2]).length for k in range(len(b) - 1)]
+            )  # Length of each segment
+        if np.any(np.array(poly_segments) > 1.0):
+            warnings.warn(
+                'Polys contain large (>1deg) segments. This could lead to unsuspected errors. To prevent errors over large region, please segmentize your polys.',
+                UserWarning,
+            )
+
```

Comment:
```suggestion
    def _check_polys_length(polys, threshold=1):
        # Check length of polys segments, issue warning if too long
        check_polys, check_holes, _, _ = split_polygons_and_holes(polys)
        check_polys.extend(check_holes)
        poly_segments = []
        for check_poly in check_polys:
            b = check_poly.boundary.coords
            # Length of each segment
            poly_segments.extend([LineString(b[k : k + 2]).length for k in range(len(b) - 1)])
        if np.any(np.array(poly_segments) > threshold):
            warnings.warn(
               f"`polys` contains large (> {threshold}°) segments. This could lead to errors over large regions. For a more accurate average, segmentize (densify) your shapes with  `shapely.segmentize(polys)`",
                UserWarning,
                stacklevel=2
            )

```
"threshold" : Simply a way to regroup the criterion in a single spot, so it may be edited later.

I also changed the warning message to explicitly mention `segmentize`.

`stacklevel=2` will make the warning appear as if it was raised from the `__init__`, instead of this method.